enumerable.rb

```diff
@@ -0,0 +1,134 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    if block_given?
+      (0..length - 1).my_each do |i|
+        yield(self[i])
+      end
+    else
+      puts "You didn't pass in a block!"
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      (0..length - 1).my_each do |i|
+        yield(self[i], i)
+      end
+    else
+      puts "You didn't pass in a block!"
+    end
+  end
+
+  def my_select
+    if block_given?
+      new_arr = []
+      (0..length - 1).my_each do |i|
+        new_arr.push(self[i]) if yield(self[i])
+      end
+    else
+      puts "You didn't pass in a block!"
+    end
+  end
+
+  def my_all?
+    (0..length - 1).my_each do |element|
+      return false unless yield(element)
+    end
+    true
+  end
```

Comment:
- We need to fix my_each first to make this method works.

- my_all? should behave the same way the original does:
`['ant', 'bear', 'cat'].all? { |word| word.length >= 3 } #=> true`
`['ant', 'bear', 'cat'].all? { |word| word.length >= 4 } #=> false`
`['ant', 'bear', 'cat'].all?(/t/)                        #=> false`
`[1, 2, 3].all?(Integer)                       #=> true`
`[nil, true, 99].all?                              #=> false`
`[].all?                                           #=> true`